### PR TITLE
add docs for RUN --ssh flag

### PR DESCRIPTION
**Pull request by alexcb**

### CODE REVIEW

File: docs/earthfile/earthfile.md [hunk 1]
```diff
@@ -185,6 +185,21 @@ release:
...
+
+Here is an example:
+
+```Dockerfile
+git-clone:
+    RUN --ssh git clone git@host.com:user/private-repo.git
+```
+
+```bash
+earth --secret GH_TOKEN="the-actual-secret-token-value" +release
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:200]: This line seems unrelated?

File: docs/earthfile/earthfile.md [hunk 2]
```diff
@@ -185,6 +185,21 @@ release:
 earth --secret GH_TOKEN="the-actual-secret-token-value" +release
 ```
 
+##### `--ssh`
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:188]: We should also add `[--ssh]` in the synopsis of `RUN` above.

File: docs/earthfile/earthfile.md (hunk 1 above)
Comment (alexcb) [reply docs/earthfile/earthfile.md:200]: ekk, that accidentally snuck in.

File: docs/earthfile/earthfile.md [hunk 3]
```diff
@@ -181,6 +181,17 @@ release:
...
+
+Allows a command to access the ssh authentication client running on the host via the socket which is referenced by the environment variable `SSH_AUTH_SOCK`.
+
+Here is an example:
+
+```Dockerfile
+git-clone:
+    RUN --ssh git clone git@host.com:user/private-repo.git
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:192]: I just realized that we give an example of an operation that we generally recommend to be done via our own `GIT CLONE` command. Perhaps replace with `go mod download`?

```suggestion
    RUN --ssh go mod download
```

**Review by vladaionescu — APPROVED** [review]

File: docs/earthfile/earthfile.md (hunk 3 above)
Comment (alexcb) [reply docs/earthfile/earthfile.md:192]: that's a really good point. If we want to use go, then we might also have to change it to:
`RUN --ssh git config --global url."git@github.com:".insteadOf "https://github.com/" && go mod download`

Comment (vladaionescu) [reply docs/earthfile/earthfile.md:192]: SGTM

**Review by vladaionescu — APPROVED** [review]
